presto-orc/src/main/java/com/facebook/presto/orc/stream/SharedBuffer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.stream;
+
+import com.facebook.presto.orc.OrcLocalMemoryContext;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import static java.util.Objects.requireNonNull;
+
+@NotThreadSafe
+public class SharedBuffer
```

Comment:
I see this class is only for reusing the compression buffer in OrcInputStream. Would it make sense to have a more generic allocator that implements `ArrayAllocator`?

Reply:
This class is designed as a single buffer that can be shared.

Ideally we would simply share a `byte[]` instance. But we need to support expansion (`ensureCapacity`) + account for memory during expansion. Thus it is much simple concept than array allocator. I would prefer to keep these two concepts separate. What do you think?